core/src/main/java/org/mule/DefaultMuleMessage.java

```diff
@@ -2111,7 +2111,6 @@ protected void copyToInbound(DefaultMuleMessage newMessage) throws Exception
         }
 
         newMessage.clearProperties(PropertyScope.INBOUND);
-        newMessage.clearProperties(PropertyScope.INVOCATION);
```

Comment:
I thought this was incorrect, but I see now it is used in VM and in this use it is correct to remove these.  I'll put this back and instead use a different method in TestMessageDispatcher.